### ISSUE TITLE
CI flake: "provided port is already allocated" occassionally happens during e2e cross cluster tests

### ISSUE THREAD

**Issue by AustinAbro321** [zarf-dev/zarf#2357]
We have seen this flake a few times, it is unclear why it is happening. 
In all of the cases the error is `provided port is already allocated` 
In the following two cases the error is on the below line. 
- https://github.com/defenseunicorns/zarf/actions/runs/8100430794/job/22138566751
- https://github.com/defenseunicorns/zarf/actions/runs/8160962762/job/22308786570
```
_, initStdErr, err := e2e.Zarf("init", "--components="+initComponents, "--nodeport", "31337", "-l", "trace", "--confirm")
require.NoError(t, err)
```
We also see a similar error here on here on test 99: 
- https://github.com/defenseunicorns/zarf/actions/runs/8012041029/job/21886586280
```
stdOut, stdErr, err = e2e.Zarf("init", "--components=k3s", "--confirm")
require.NoError(t, err, stdOut, stdErr)
```

**Comment by AustinAbro321** [comment]
We believe that during the first two occurrences of the the problem was the registry port 31337 assigned in the test conflicted with the randomly generated trafeik ports created by k3d. This will be solved by disabling trafeik during e2e tests in #2282. 

The third issue is an actual bug in zarf. It is caused by the fact that the injector (seed registry) selects a random port and this port could conflict with the registry port. Looking at the logs here confirms this as the seed registry matches the registry in the logs https://github.com/defenseunicorns/zarf/actions/runs/8012041029/job/21886586280.

**Comment by lucasrod16** [comment]
Hit this again: https://github.com/defenseunicorns/zarf/actions/runs/8841615794/job/24279036176?pr=2456#step:7:494

**Comment by AustinAbro321** [comment]
Yeah it's only like a 1 in 3000 chance, but it's still worth fixing for the init case. Especially since this can also happen to real uses if they have the proper luck or lack thereof

**Comment by AustinAbro321** [comment]
I'm thinking about the best way to solve this. I think there are few reasonable paths: 
1. We could use the [portAllocator](https://github.com/kubernetes/kubernetes/blob/bc8491416d5279494ef0cceed97cf68ed00d8cab/pkg/registry/core/service/portallocator/allocator.go#L17) package in Kubernetes. This seems quite complex to setup, though likely we could come to a clean solution.
1. We could create the injector service, check if it has the same port as the nodeport service then delete and re-create it until it doesn't match
1. We could generate a random nodeport number in the range. If any service or nodeport is using a number that number we can re-generate the number until it's safe. A potential downside is that we need to figure out the users nodeport range. 

I'm partial to option 2. While it's a bit of a hack, it is very low cost and simple to implement and won't require additional dependencies.
